fullstack-cert/python-projects/matrix-converter-lab/main.py

```diff
@@ -0,0 +1,22 @@
+def adjacency_list_to_matrix(adj_list):
+    n = len(adj_list)
+    
+    adj_matrix = [[0] * n for _ in range(n)]
+
+    for src_node, neighbors in adj_list.items(): 
+        for dest_node in neighbors:
+            adj_matrix[src_node][dest_node] = 1
+
+    for row in adj_matrix:
+        print(row)
+
+    return adj_matrix
+
+adj_list = {
+    0: [1, 2],
+    1: [2],
+    2: [0, 3],
+    3: [2]
+}
```

Comment:
is it intended that some of these connections are unidirectional?

Reply:
What about `The function should take a dictionary representing the adjacency list of an unweighted (either undirected or directed) graph as its argument.`?

Reply:
that would be fine